pyvista/core/dataset.py

```diff
@@ -233,30 +392,61 @@ def arrows(self) -> Optional['pyvista.PolyData']:
         """Return a glyph representation of the active vector data as arrows.
 
         Arrows will be located at the points of the mesh and
-        their size will be dependent on the length of the vector.
+        their size will be dependent on the norm of the vector.
         Their direction will be the "direction" of the vector
 
         Returns
         -------
-        arrows : pyvista.PolyData
-            Active scalars represented as arrows.
+        :class:`pyvista.PolyData`
+            Active vectors represented as arrows.
+
+        Examples
+        --------
+        Create a mesh, compute the normals and set them active, and
+        plot the active vectors.
+
+        >>> import pyvista
+        >>> mesh = pyvista.Cube().clean()
+        >>> mesh_w_normals = mesh.compute_normals()
+        >>> mesh_w_normals.active_vectors_name = 'Normals'
+        >>> arrows = mesh_w_normals.arrows
+        >>> arrows.plot(show_scalar_bar=False)
 
         """
-        name = self.active_vectors_name
-        return None if name is None else self.glyph(scale=name, orient=name)
+        vectors_name = self.active_vectors_name
+        if vectors_name is None:
+            return
+
+        if self.active_vectors.ndim != 2:  # type: ignore
+            raise ValueError('Active vectors are not vectors.')
+
+        scale_name = f'{vectors_name} Magnitude'
+        scale = np.linalg.norm(self.active_vectors, axis=1)
+        self.point_arrays.set_array(scale, scale_name)
+        return self.glyph(orient=vectors_name, scale=scale_name)
 
     @property
-    def vectors(self) -> Optional[pyvista_ndarray]:
-        """Return active vectors."""
+    def vectors(self) -> Optional[pyvista_ndarray]:  # pragma: no cover
+        """Return active vectors.
+
+        .. deprecated:: 0.31.0
```

Comment:
Is this version correct? I thought it was more recent. Same for the setter.

Reply:
Deprecated in #1461, which is after 0.31.0.  Correct version should be 0.32.0

```suggestion
        .. deprecated:: 0.32.0
```